### PR TITLE
Functions Yaml File Additions

### PR DESCRIPTION
**Pull request by nirpechuk**
## Changes
- Added cross-links to all the functions yaml files
- Added sections to presto functions (did not change the kinds)

## Testing/Review
I tested to make sure the docs can still be generated and that I didn't break anything to do with docs, but I would appreciate review to make sure that it doesn't break anything else. Thank you!

### CODE REVIEW

File: src/trino/btrim_impl.rs [hunk 1]
```diff
@@ -0,0 +1,143 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```
Comment (wizardxz) [inline src/trino/btrim_impl.rs:1]: Why these files are added?

Comment (nirpechuk) [reply src/trino/btrim_impl.rs:1]: Probably a merge mistake--should I remove those changes?

File: assets/trino/functions.sdf.yml [hunk 2]
```diff
@@ -1233,6 +1489,7 @@ function:
   returns:
     datatype: varchar
   implemented-by: !datafusion
+  cross-link: https://prestodb.io/docs/current/functions.html
```
Comment (vgapeyev) [inline assets/trino/functions.sdf.yml:1492]: Add `section: string` to the two `concat_ws` entries?

File: assets/trino/functions.sdf.yml [hunk 3]
```diff
@@ -1263,6 +1524,7 @@ function:
   optional-parameters: []
   returns:
     datatype: boolean
+  cross-link: https://prestodb.io/docs/current/functions.html
```
Comment (vgapeyev) [inline assets/trino/functions.sdf.yml:1527]: `array` section?

File: assets/trino/functions.sdf.yml [hunk 4]
```diff
@@ -3409,6 +4133,8 @@ function:
   optional-parameters: []
   returns:
     datatype: double
+  cross-link: https://prestodb.io/docs/current/functions/logical.html#mod
+  section: logical
```
Comment (vgapeyev) [inline assets/trino/functions.sdf.yml:4137]: `mod` ought to be in `math`, both in the section and the URL?

File: assets/trino/functions.sdf.yml [hunk 5]
```diff
@@ -3582,6 +4344,8 @@ function:
   returns:
     datatype: $1
   implemented-by: !datafusion
+  cross-link: https://prestodb.io/docs/current/functions/conditional.html#nullif
+  section: scalar
```
Comment (vgapeyev) [inline assets/trino/functions.sdf.yml:4348]: Isn't section basically taken out of the URL, so `conditional` here?

File: assets/trino/functions.sdf.yml [hunk 6]
```diff
@@ -3647,6 +4421,7 @@ function:
   optional-parameters: []
   returns:
     datatype: timestamp
+  cross-link: https://prestodb.io/docs/current/functions.html
```
Comment (vgapeyev) [inline assets/trino/functions.sdf.yml:4424]: @wizardxz , this `objectid_timestamp` (and probably `objectid` above it) seem to be specific to the MongoDB connector.  Should we drop them from the list for this reason? 
See https://trino.io/docs/current/connector/mongodb.html#objectid_timestamp

File: assets/trino/functions.sdf.yml [hunk 7]
```diff
@@ -3697,6 +4482,8 @@ function:
   returns:
     datatype: double
   implemented-by: !datafusion
+  cross-link: https://prestodb.io/docs/current/functions/logical.html#pi
+  section: logical
```
Comment (vgapeyev) [inline assets/trino/functions.sdf.yml:4486]: Another `math` vs `logical`.

File: assets/trino/functions.sdf.yml [hunk 8]
```diff
@@ -4123,6 +4995,8 @@ function:
   returns:
     datatype: boolean
   implemented-by: !rust
+  cross-link: https://prestodb.io/docs/current/functions/lambda.html#regexp_like
+  section: lambda
```
Comment (vgapeyev) [inline assets/trino/functions.sdf.yml:4999]: `lambda` vs `regexp`

File: assets/trino/functions.sdf.yml [hunk 9]
```diff
@@ -4675,6 +5664,8 @@ function:
   optional-parameters: []
   returns:
     datatype: real
+  cross-link: https://prestodb.io/docs/current/functions/comparison.html#sign
+  section: comparison
```
Comment (vgapeyev) [inline assets/trino/functions.sdf.yml:5668]: `math` rather than `comparison`

File: assets/trino/functions.sdf.yml [hunk 10]
```diff
@@ -6418,54 +7784,97 @@ function:
...
+function:
+  name: date_part
+  parameters:
+  - datatype: varchar
+  - datatype: date
+  optional-parameters: []
+  returns:
+    datatype: double
+  implemented-by: !datafusion
+  cross-link: https://prestodb.io/docs/current/functions.html
```
Comment (vgapeyev) [inline assets/trino/functions.sdf.yml:7877]: Another unfortunate merge conflict?  `date_part` got moved out of here.

File: assets/trino/functions.sdf.yml [hunk 11]
```diff
@@ -6377,25 +7735,31 @@ function:
   optional-parameters: []
   returns:
     datatype: array<$9>
+  cross-link: https://prestodb.io/docs/current/functions/array.html#zip_with
+  section: array
 ---
```
Comment (vgapeyev) [inline assets/trino/functions.sdf.yml:7740]: Actually, based on the alphabetic ordering (`zip_with` above, `if` below) I suspect the whole rest of the file is a victim of an unfortunate merge.

**Review by vgapeyev — COMMENTED** [review]
I scrolled through `trino/functions.sdf.yml` and there are a few places with a missing `section` or a mis-tagged section and URL. I commented only on some of them.

**Pull request closed without merging**